### ISSUE TITLE
Git check-ignore should gracefully handle non-zero exit codes

### ISSUE THREAD

**Issue by westse** [specmatic/specmatic#861]
**Description**
This already happens for exit code 1, which is interpreted as .gitignore needing to be updated. I believe this should happen for any non-zero exit code.

**Steps to reproduce**
1. I believe this is system dependent, so see System info below

**Expected behavior**
check-ignore should gracefully handle non-zero exit codes and interpret it as .gitignore needing to be updated.

**System Information:**
 - OS & version: [e.g. Ubuntu 22.04]
 - Specmatic version: [e.g. 1.0.4]
 - JDK version: [e.g. 17]

**Comment by joelrosario** [comment]
@westse Thanks for reporting this, and raising a PR to go with it!

Quick question: The documentation for git check-ignore describes the following exit codes:
0 - the path is ignored
1 - the path is not ignore OR .gitignore does not exist
128 - anything else

We have tested on a Mac, and were able to trigger all 3 conditions (we ran `git check-ignore` outside a git directory and got 128).

Could you share a screenshot of the error you are seeing please.

Thanks!

**Comment by westse** [comment]
@joelrosario I'm on Ubuntu 22.04 and it looks like my experience mostly matches yours with one addition (exitCode=127):
- 0 - the path is ignored
- 1 - the path is not ignore OR .gitignore does not exist
- 128 - Running `git` outside a git working directory.
- 127 - `git` not installed.

I'm now not sure where I saw the exitCode=2 - I'll remove it from the issue description.

Some more detail if it is helpful:
That last scenario (no system `git` installed) was actually how I ran previously (in a container), which was possible since Specmatic's git clone operation falls back to JGit. But it became no longer possible starting with version 0.81.2 because the new `check-ignore` command does *not* fall back to JGit. So I then I tried ensuring a system `git` was installed and recalled why I didn't use it before: I was blocked by #863 (I see this is fixed now that you merged my PR, thanks). If I recall correctly, even with #863 fixed I was still seeing `check-ignore` fail with non-zero/one code (probably 128). Hope that helps.
